include/seqan3/core/algorithm/concept.hpp

```diff
@@ -24,7 +24,7 @@ namespace seqan3
 {
 //!\cond
 // Forward declarations
-template <typename derived_t, typename value_t>
+template <typename derived_t, typename value_t, bool needs_value = false>
```

Comment:
We said we wanted to add `has_value_member` which defaults to `true` and that specifying `true` is the new behaviour; but this can be of course done one way or the other :)

The actual point I want to make is, that if you default it here to `true`, you don't need to touch every configuration object (they will just default to `true` and have the member).
You would only need to change it for the new ones - which I think is preferable, because there are already some PRs that refactor configs and they would all get merge conflicts.

Reply:
I thought the other way round: If now we add the additional parameter, then each PR will remove them gradually, and we reach the final design without further API breaks. In the end, we can simply remove the version with 3 template parameters. 

If we default to `true` now, then we have to remove all the additional parameters in another big API change that touches every config object. Are you sure this is the way we want to go? If yes, then I'll revert my changes tomorrow and apply your suggestion.

Reply:
My original intent was to not touch existing code, because everyone who touches will use the `false` one. If everyone uses `false` we can remove the extra.

Your approach is exactly the same, but in reverse. I don't mind this works equally as good. And I think it is the same amount of work.

----

BTW

Oh I haven't seen that the second template parameter is not needed anymore after this.
```suggestion
template <typename derived_t, typename value_t = void, bool needs_value = false>
```
so we need to do this :)

```cpp
struct hit_strata : public pipeable_config_element<hit_strata, uint8_t, true>

// will become after adaptation

struct hit_strata : public pipeable_config_element<hit_strata>
```